app/scripts/lib/logger.js

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Shared logging module that performs the checks and string
+ * interpolation.
+ *
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  /**
+   * Constructor of log module.
+   *
+   * @param logWindow Window object that contains console used for logging.
+   *
+   * @constructor
+   */
+  function Logger(logWindow) {
+    this._window = logWindow || window;
+  }
+
+  Logger.prototype = {
+
+    /**
+     * Wrapper for `console.info` function that checks for availability and
+     * then prints info.
+     *
+     */
+    info: function () {
+      if (this._window.console && this._window.console.info) {
```

Comment:
Not a big deal and definitely a question rather than a suggestion: What if we did these checks once, during construction/initialisation, rather than every time a log method is invoked? If the check fails on initialisation, we can just turn the methods into nops.